examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt

```diff
@@ -0,0 +1,160 @@
+package com.mapbox.navigation.examples.core
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import com.mapbox.navigation.ui.map.NavigationMapboxMapInstanceState
+import kotlinx.android.synthetic.main.free_drive_navigation_layout.*
+import timber.log.Timber
+
+/**
+ * To ensure proper functioning of this example make sure your Location is turned on.
+ */
+class FreeDriveNavigationActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    companion object {
+        const val MAP_INSTANCE_STATE_KEY = "navgation_mapbox_map_instance_state"
+        const val DEFAULT_INTERVAL_IN_MILLISECONDS = 1000L
+        const val DEFAULT_MAX_WAIT_TIME = DEFAULT_INTERVAL_IN_MILLISECONDS * 5
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private var mapInstanceState: NavigationMapboxMapInstanceState? = null
```

Comment:
Given this is an example activity shouldn't they be kept to a minimum so that the code that's being demonstrated stands out? I wouldn't normally have these as nullable variables but if I implement my own way of handling this condition and everybody else does it their way then there won't be any consitency

Reply:
That's fair but I'd showcase examples the way we think is 💅 for customer. What are we trying to show here? That those can be nullable and how to implement with that constraint in mind? I'd rather do the opposite and try to "educate" clients of using our components with "the" non-null approach. But yeah, totally agreed with your concern on consistency in the examples. We should discuss this as a team and apply same approach across all examples 👍 Does that sound reasonable?

Reply:
It's definitely a balance to decide how opinionated the examples should be. We as a group should come to a consensus.